stix2/pattern_expressions.py

```diff
@@ -0,0 +1,223 @@
+from .constants import Constant, IntegerConstant, ListConstant, make_constant
+from .object_path import ObjectPath
+
+
+class PatternExpression(object):
+
+    @staticmethod
+    def escape_quotes_and_backslashes(s):
+        return s.replace(u'\\', u'\\\\').replace(u"'", u"\\'")
+
+
+class ComparisonExpression(PatternExpression):
+    def __init__(self, operator, lhs, rhs, negated=False):
```

Comment:
You don't create instances of ComparisonExpression, but EqualityComparisonExpression, MatchesComparisonExpression, etc.  This is to enable the PatternMatcher to have different functionality for each type of ComparisonExpression - assuming we start using this stuff in Andy's code...